model/src/main/proto/oppia_logger.proto

```diff
@@ -21,46 +21,66 @@ message EventLog {
     oneof activity_context {
       // The event being logged is related to exploration.
       ExplorationContext exploration_context = 1;
+
       // The event being logged is related to the question player.
       QuestionContext question_context = 2;
+
       // The event being logged is related to the topic activity.
       TopicContext topic_context = 3;
+
       // The event being logged is related to the story activity.
       StoryContext story_context = 4;
+
       // The event being logged is related to the concept card.
       ConceptCardContext concept_card_context = 5;
+
       // The event being logged is related to the revision card.
       RevisionCardContext revision_card_context = 6;
+
       // The event being logged is related to starting an exploration card.
-      StartCardContext start_card_context = 7;
+      CardContext start_card_context = 7;
+
       // The event being logged is related to ending an exploration card.
-      EndCardContext end_card_context = 8;
+      CardContext end_card_context = 8;
+
       // The event being logged is related to offering a hint when it becomes available.
-      HintOfferedContext hint_offered_context = 9;
+      HintContext hint_offered_context = 9;
+
       // The event being logged is related to accessing a hint.
-      AccessHintContext access_hint_context = 10;
+      HintContext access_hint_context = 10;
+
       // The event being logged is related to offering a solution when it becomes available.
-      SolutionOfferedContext solution_offered_context = 11;
+      ExplorationDetailsContext solution_offered_context = 11;
```

Comment:
Hmm this is really close in name to ExplorationContext. Is there any way to consolidate the two?

Reply:
clubbed into one, we can handle packing and unpacking of data from controller and utility separately.